packages/realm/src/tests/schema-normalization.test.ts

```diff
@@ -234,6 +248,22 @@ describe("normalizePropertySchema", () => {
       "linkingObjects",
       "To define an inverse relationship, use { type: 'linkingObjects', objectType: 'MyObjectType', property: 'myObjectTypesProperty' }",
     );
+
+    // -------------------------
+    // Indexed & Primary Keys
+    // -------------------------
+
+    itThrowsWhenNormalizing(
+      "string?",
+      "Optional properties cannot be used as a primary key.",
+      true, // Declare that it is a primary key.
+    );
```

Comment:
Incoming personal opinion:
For a third boolean positional argument, that you feel a need to provide a comment for, like here, it's often better for the entire function to take a single object argument or a third object argument, with a named parameter - making the code speak for itself:

```suggestion
    itThrowsWhenNormalizing(
      "string?",
      "Optional properties cannot be used as a primary key.",
      { isPrimary: true },
    );
```

Reply:
Amazing feedback, thank you!